crates/turbopack-ecmascript/src/tree_shake/graph.rs

```diff
@@ -482,19 +493,23 @@ impl DepGraph {
                 .count();
 
             if dependant_count >= 2 && count_of_startings >= 2 {
-                groups.push((vec![id.clone()], FxHashSet::default()));
+                groups.push((vec![id.clone()], FxHashSet::default(), 1));
                 global_done.insert(ix as u32);
             }
         }
 
         loop {
             let mut changed = false;
 
-            for (group, group_done) in &mut groups {
-                let start = &group[0];
-                let start_ix = self.g.get_node(start);
-                changed |=
-                    add_to_group(&self.g, data, group, start_ix, &mut global_done, group_done);
+            for (group, group_done, init_len) in &mut groups {
```

Comment:
I'm not sure why `init_len` was added here when it always ends up getting set to `group.len()`? Can we just do `for i in 0..group.len()` or `for start in &group`?

Reply:
`group` is modified in the loop so we have to track the initial length of the `group`